homeassistant/components/telegram_bot/__init__.py

```diff
@@ -503,12 +505,18 @@ def process_message(self, data):
             if event_data is None:
                 return False
 
-            if data[ATTR_TEXT][0] == '/':
-                pieces = data[ATTR_TEXT].split(' ')
-                event_data[ATTR_COMMAND] = pieces[0]
-                event_data[ATTR_ARGS] = pieces[1:]
+            if 'text' in data:
+                if data['text'][0] == '/':
+                    pieces = data['text'].split(' ')
+                    event_data[ATTR_COMMAND] = pieces[0]
+                    event_data[ATTR_ARGS] = pieces[1:]
+                else:
+                    event_data[ATTR_TEXT] = data['text']
+                    event = EVENT_TELEGRAM_TEXT
             else:
-                event_data[ATTR_TEXT] = data[ATTR_TEXT]
+                # Some other thing...
+                _LOGGER.warning('SOME OTHER THING RECEIVED --> "%s"', data)
```

Comment:
Clarify to something like:
```py
_LOGGER.warning("Message without text data received: %s", data)
```
Note the use of quotes. That's home assistant standard now.